Misleading name of function: `get_amino_acids_percent()` returns not percentages but fractions

### Setup

I am reporting a problem with Biopython version, Python version, and operating system as follows:

```python
>>> import sys; print(sys.version)
3.8.10 (default, Mar 25 2024, 10:42:49)
[GCC 9.4.0]
>>> import sys; print(sys.version)
3.8.10 (default, Mar 25 2024, 10:42:49)
[GCC 9.4.0]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-5.11.0-1022-aws-x86_64-with-glibc2.29
>>> import Bio; print(Bio.__version__)
1.83
```

And this is a code I tried to see behavior. I suggest either changing the function name to `get_amino_acids_fraction()` or modifying the return value of the `get_amino_acids_percent()` function to be a percentage in the range of 0 to 100.

```python
from Bio.SeqUtils.ProtParam import ProteinAnalysis

X = ProteinAnalysis("MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGT"
                    "RDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEEC"
                    "LFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILF"
                    "LPLPV")

sum_pct = 0.0
pct_aa = X.get_amino_acids_percent()
for aa, pct in pct_aa.items():
    print(aa, pct, sep="\t")
    sum_pct += pct
print(f"sum_pct: {sum_pct.1f}")
```

### Expected behaviour

```
A       3.9473684210526314
C       1.9736842105263157
D       3.289473684210526
E       7.894736842105263
F       3.9473684210526314
G       9.210526315789473
H       3.289473684210526
I       3.289473684210526
K       7.894736842105263
L       11.842105263157894
M       1.3157894736842104
N       4.605263157894736
P       5.263157894736842
Q       3.9473684210526314
R       3.9473684210526314
S       6.578947368421052
T       8.552631578947368
V       3.289473684210526
W       0.6578947368421052
Y       5.263157894736842
sum_pct: 100.0
```

### Actual behaviour

```
A       0.039473684210526314
C       0.019736842105263157
D       0.03289473684210526
E       0.07894736842105263
F       0.039473684210526314
G       0.09210526315789473
H       0.03289473684210526
I       0.03289473684210526
K       0.07894736842105263
L       0.11842105263157894
M       0.013157894736842105
N       0.046052631578947366
P       0.05263157894736842
Q       0.039473684210526314
R       0.039473684210526314
S       0.06578947368421052
T       0.08552631578947369
V       0.03289473684210526
W       0.006578947368421052
Y       0.05263157894736842
sum_pct: 1.0
```

--- a/Bio/SeqUtils/ProtParam.py
+++ b/Bio/SeqUtils/ProtParam.py
@@ -17,9 +17,9 @@ Examples
 >>> print(X.count_amino_acids()['E'])
 12
 >>> print("%0.2f" % X.get_amino_acids_percent()['A'])
-0.04
+3.95
 >>> print("%0.2f" % X.get_amino_acids_percent()['L'])
-0.12
+11.84
 >>> print("%0.2f" % X.molecular_weight())
 17103.16
 >>> print("%0.2f" % X.aromaticity())

--- a/Bio/SeqUtils/ProtParam.py
+++ b/Bio/SeqUtils/ProtParam.py
@@ -16,9 +16,9 @@ Examples
 6
 >>> print(X.count_amino_acids()['E'])
 12
->>> print("%0.2f" % X.get_amino_acids_percent()['A'])
+>>> print("%0.2f" % X.amino_acids_percent['A'])
 3.95
->>> print("%0.2f" % X.get_amino_acids_percent()['L'])
+>>> print("%0.2f" % X.amino_acids_percent['L'])
 11.84
 >>> print("%0.2f" % X.molecular_weight())
 17103.16
@@ -49,8 +49,11 @@ Other public methods are:
 
 """
 
+import functools
 import sys
+import warnings
 
+from Bio import BiopythonDeprecationWarning
 from Bio.Data import IUPACData
 from Bio.Seq import Seq
 from Bio.SeqUtils import IsoelectricPoint  # Local
@@ -77,7 +80,6 @@ class ProteinAnalysis:
         """Initialize the class."""
         self.sequence = prot_sequence.upper()
         self.amino_acids_content = None
-        self.amino_acids_percent = None
         self.length = len(self.sequence)
         self.monoisotopic = monoisotopic
 
@@ -100,26 +102,32 @@ class ProteinAnalysis:
         return self.amino_acids_content
 
     def get_amino_acids_percent(self):
-        """Calculate the amino acid content in percentages.
+        """Included for backwards compatibility (DEPRECATED)."""
+        warnings.warn(
+            "The get_amino_acids_percent method has been deprecated "
+            "and will likely be removed from Biopython in the near "
+            "future. Please use the amino_acids_percent attribute instead.",
+            BiopythonDeprecationWarning,
+        )
+
+        return {aa: percent / 100 for aa, percent in self.amino_acids_percent.items()}
+
+    @functools.cached_property
+    def amino_acids_percent(self):
+        """Get the amino acid content in percentages.
 
         The same as count_amino_acids only returns the Number in percentage of
         entire sequence. Returns a dictionary of {AminoAcid:percentage}.
 
-        The return value is cached in self.amino_acids_percent.
-
-        input is the dictionary self.amino_acids_content.
-        output is a dictionary with amino acids as keys.
+        Unlike the deprecated get_amino_acids_percent method, this attribute
+        returns percentages in the range 0-100.
         """
-        if self.amino_acids_percent is None:
-            aa_counts = self.count_amino_acids()
+        aa_counts = self.count_amino_acids()
+        percentages = {
+            aa: (count * 100 / self.length) for aa, count in aa_counts.items()
+        }
 
-            percentages = {
-                aa: (count * 100 / self.length) for aa, count in aa_counts.items()
-            }
-
-            self.amino_acids_percent = percentages
-
-        return self.amino_acids_percent
+        return percentages
 
     def molecular_weight(self):
         """Calculate MW from Protein sequence."""
@@ -134,7 +142,7 @@ class ProteinAnalysis:
         It is simply the relative frequency of Phe+Trp+Tyr.
         """
         aromatic_aas = "YWF"
-        aa_percentages = self.get_amino_acids_percent()
+        aa_percentages = self.amino_acids_percent
 
         aromaticity = sum(aa_percentages[aa] / 100 for aa in aromatic_aas)
 
@@ -331,7 +339,7 @@ class ProteinAnalysis:
 
         Returns a tuple of three floats (Helix, Turn, Sheet).
         """
-        aa_percentages = self.get_amino_acids_percent()
+        aa_percentages = self.amino_acids_percent
 
         helix = sum(aa_percentages[r] / 100 for r in "EMALK")
         turn = sum(aa_percentages[r] / 100 for r in "NPGSD")

--- a/Bio/SeqUtils/ProtParam.py
+++ b/Bio/SeqUtils/ProtParam.py
@@ -113,7 +113,9 @@ class ProteinAnalysis:
         if self.amino_acids_percent is None:
             aa_counts = self.count_amino_acids()
 
-            percentages = {aa: count / self.length for aa, count in aa_counts.items()}
+            percentages = {
+                aa: (count * 100 / self.length) for aa, count in aa_counts.items()
+            }
 
             self.amino_acids_percent = percentages
 
@@ -134,7 +136,7 @@ class ProteinAnalysis:
         aromatic_aas = "YWF"
         aa_percentages = self.get_amino_acids_percent()
 
-        aromaticity = sum(aa_percentages[aa] for aa in aromatic_aas)
+        aromaticity = sum(aa_percentages[aa] / 100 for aa in aromatic_aas)
 
         return aromaticity
 
@@ -331,9 +333,9 @@ class ProteinAnalysis:
         """
         aa_percentages = self.get_amino_acids_percent()
 
-        helix = sum(aa_percentages[r] for r in "EMALK")
-        turn = sum(aa_percentages[r] for r in "NPGSD")
-        sheet = sum(aa_percentages[r] for r in "VIYFWLT")
+        helix = sum(aa_percentages[r] / 100 for r in "EMALK")
+        turn = sum(aa_percentages[r] / 100 for r in "NPGSD")
+        sheet = sum(aa_percentages[r] / 100 for r in "VIYFWLT")
 
         return helix, turn, sheet
 

--- a/DEPRECATED.rst
+++ b/DEPRECATED.rst
@@ -521,6 +521,9 @@ was updated to be consistent with the calculated values by Sharp & Li.
 Function 'GC' in Bio.SeqUtils was deprecated in Release 1.80, and removed in
 Release 1.82. Instead use function 'gc_fraction'.
 
+Function get_amino_acids_percent in Bio.SeqUtils.ProteinAnalysis was deprecated
+in Release 1.85. Use the amino_acids_percent property instead.
+
 Bio.PopGen.Async
 ----------------
 ``Bio.PopGen.Async`` was deprecated in Release 1.68, removed in Release 1.70.

--- a/Tests/test_ProtParam.py
+++ b/Tests/test_ProtParam.py
@@ -9,6 +9,7 @@
 
 import unittest
 
+from Bio import BiopythonDeprecationWarning
 from Bio.Seq import Seq
 from Bio.SeqRecord import SeqRecord
 from Bio.SeqUtils import molecular_weight
@@ -39,13 +40,24 @@ class ProtParamTest(unittest.TestCase):
                 self.assertEqual(count_dict[i], self.text.count(i))
 
     def test_get_amino_acids_percent(self):
+        """Calculate amino acid percentages (DEPRECATED)."""
+        with self.assertWarns(BiopythonDeprecationWarning):
+            for analysis in self.analyses:
+                percent_dict = analysis.get_amino_acids_percent()
+                seq_len = len(self.text)
+                for i in percent_dict:
+                    self.assertAlmostEqual(
+                        percent_dict[i], self.text.count(i) / seq_len
+                    )
+
+    def test_amino_acids_percent(self):
         """Calculate amino acid percentages."""
         for analysis in self.analyses:
-            percent_dict = analysis.get_amino_acids_percent()
             seq_len = len(self.text)
-            for i in percent_dict:
+            for i in analysis.amino_acids_percent:
                 self.assertAlmostEqual(
-                    percent_dict[i], (self.text.count(i) * 100 / seq_len)
+                    analysis.amino_acids_percent[i],
+                    (self.text.count(i) * 100 / seq_len),
                 )
 
     def test_get_molecular_weight(self):

--- a/Tests/test_ProtParam.py
+++ b/Tests/test_ProtParam.py
@@ -44,7 +44,9 @@ class ProtParamTest(unittest.TestCase):
             percent_dict = analysis.get_amino_acids_percent()
             seq_len = len(self.text)
             for i in percent_dict:
-                self.assertAlmostEqual(percent_dict[i], self.text.count(i) / seq_len)
+                self.assertAlmostEqual(
+                    percent_dict[i], (self.text.count(i) * 100 / seq_len)
+                )
 
     def test_get_molecular_weight(self):
         """Calculate protein molecular weight."""

